src/aiokatcp/sensor.py

```diff
@@ -187,6 +188,37 @@ def __init__(
         self.auto_strategy_parameters = tuple(auto_strategy_parameters)
         # TODO: should validate the parameters against the strategy.
 
+    def _check_value_type(self, value: _T) -> _T:
+        """Verify incoming value is compatible with sensor type.
+
+        This also handles the special case of :class:`core.Timestamp` where
+        it is used with `float` type interchangeably.
+
+        Raises
+        ------
+        TypeError
+            If the incoming `value` type is not compatible with the sensor's
+            core type.
+        """
+
+        if isinstance(value, self._core_type) and not issubclass(self._core_type, enum.Enum):
+            # The more general case of e.g. numpy types not being directly
+            # comparable with python types.
+            return value
+        elif self.stype is core.Timestamp and isinstance(value, numbers.Real):
+            # core.Timestamp can also be a float
+            return value  # type: ignore
+        elif type(value) in [bytes, str, core.Address] and self.stype is core.Address:
```

Comment:
Aha, I think you are quite right Re
> core.Address requires an IP address object, not a string

I think I was getting all sorts of disoriented when looking at examples vs core.Address class declaration.
Still, fair point. Let me trace it (and usages) again so I can confirm this just needs to be updated to something like
```python
elif isinstance(value, _IPAddress) and self.stype is core.Address
```

I should also make sure the logical checks are at least somewhat uniform to e.g. `isinstance(...)` first.

Reply:
```
elif isinstance(value, _IPAddress) and self.stype is core.Address
```
I'm not convinced we should try to support that either. I don't think setting a sensor value to an IPv4Address has ever worked, since it's not a registered type.

Reply:
Quite right once again. Looking back at a `katsdpcontroller/master_controller.py::Product::connect`
```python
def connect(
    ...,
    host: _IPAddress,
    ...,
):
    ...
    <address_sensor>.value = Address(host, port)
```

So the value itself is provided as an instance of `core.Address`. I see a similar usage in `katsdpcontroller/tasks.py` in setting the `endpoint_sensor`, so I will work on that assumption.